ui/src/views/Roles/ViewRoles/index.jsx

```diff
@@ -16,10 +17,18 @@ import HelpView from '../../../components/HelpView';
 }))
 export default class ViewRoles extends PureComponent {
   state = {
-    roleSearch: '',
+    roleSearch: this.props.history.location.search
+      ? parse(this.props.history.location.search.slice(1)).roleSearch
+      : '',
   };
 
   handleRoleSearchSubmit = roleSearch => {
+    if (roleSearch !== this.state.roleSearch) {
+      this.props.history.push(
+        roleSearch.length > 0 ? `?${stringify({ roleSearch })}` : '/auth/roles'
+      );
+    }
+
     this.setState({ roleSearch });
   };
 
```

Comment:
STR:

1. Navigate to http://localhost:5080/auth/roles?roleSearch=foo

Expected:

<img width="258" alt="Screen Shot 2020-04-15 at 3 00 08 PM" src="https://user-images.githubusercontent.com/3766511/79377569-e32d3980-7f29-11ea-803f-98c23be4d8cf.png">

Actual:

<img width="269" alt="Screen Shot 2020-04-15 at 3 00 02 PM" src="https://user-images.githubusercontent.com/3766511/79377571-e3c5d000-7f29-11ea-8249-5d8d09a367b2.png">


This applies to the the other view as well.